### ISSUE TITLE
Low performance on querying data

### ISSUE THREAD

**Issue by nhim175** [js-data/js-data-firebase#34]
It seems like when we do a `findAll` with some query conditions, it will load the whole data from that collection and filter them on the client side. This is not scalable. 

I know this is because Firebase doesn't support complex query. Therefore I don't recommend using this adapter...

**Comment by jmdobry** [comment]
Yeah, right now the Firebase adapter is great for use cases where you have a small amount of data, such as app settings, user profiles, etc. Not great for data that grows a lot. There might be some improvements that can be made to this adapter to improve query time, I'll investigate.
